tests/core/arch-x86/stalker-x86-exceptions.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2021 Ole André Vadla Ravnås <oleavr@nowsecure.com>
+ *
+ * Licence: wxWindows Library Licence, Version 3.1
+ */
+
+#include <glib.h>
+
+extern "C"
+{
+  void test_check_bit (guint32 * val, guint8 bit);
+  void test_set_bit (const char * func, guint32 * val, guint8 bit);
+  void test_try_and_catch (guint32 * val);
+  void test_try_and_dont_catch (guint32 * val);
+}
+
+class TestException;
+class FakeException;
+class TestResource;
+
+static void test_try_and_catch_pp (guint32 * val);
+static void test_try_and_dont_catch_pp (guint32 * val);
+static void test_try_and_dont_catch_pp_2 (guint32 * val);
+static void test_try_and_dont_catch_pp_3 (guint32 * val);
+
+extern "C"
+{
+  void
+  test_check_bit (guint32 * val,
+                  guint8 bit)
+  {
+    g_assert_true ((*val & ((guint32) 1 << bit)) != 0);
+    *val &= ~(guint32) 1 << bit;
```

Comment:
```suggestion
    *val &= ~1U << bit;
```

Here and below I think we can do this instead of casting.